File: LSSEventListenerExample/src/test/java/TestListeners/TestListenerTimeWindow.java
package TestListeners;

import com.espertech.esper.common.client.EventBean;
import com.espertech.esper.runtime.client.EPRuntime;
import com.espertech.esper.runtime.client.EPStatement;
import com.espertech.esper.runtime.client.UpdateListener;
import com.espertech.EventTypes.Types.ComplexEvent;

import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;

public class TestListenerTimeWindow implements UpdateListener {
    public List<Object> emittedList = new LinkedList<>();

    @Override
    public void update(EventBean[] newEvents, EventBean[] oldEvents, EPStatement statement, EPRuntime runtime) {
        emittedList.clear();
        for (EventBean newEvent : newEvents) {
            var event = ((HashMap) newEvent.getUnderlying());
            var events = (ComplexEvent[])event.get("EventTypes");
            emittedList.addAll(List.of(events));
        }
    }
}

File: Nesper/EventAnalytics.Tests/TestListeners/TestListenerTimeWindow.cs
using com.espertech.esper.compat.collections;
using com.espertech.esper.runtime.client;
using EventAnalytics.Common.Events;

namespace EventAnalytics.Tests.TestListeners;

internal class TestListenerTimeWindow : UpdateListener
{
    public List<ComplexEvent> EmittedList = new();

    public void Update(object sender, UpdateEventArgs eventArgs)
    {
        EmittedList.Clear();
        foreach (var newEvent in eventArgs.NewEvents)
        {
            var @event = (HashMap<string, object>)newEvent.Underlying;
            var events = (ComplexEvent[])@event.Get("events");
            EmittedList.AddRange(events);
        }
    }
}
